src/content/docs/en/guides/backend/google-firebase.mdx

```diff
@@ -11,6 +11,432 @@ import FileTree from '~/components/FileTree.astro'
+- Firebase credentials: You will need two sets of credentials to connect Astro to Firebase:
+  - Web app credentials: These credentials will be used by the client side of your app. You can find them in the Firebase console under Project settings > General, scroll down to the "Your apps" section and click on the "Web app" icon.
+  - Project credentials: These credentials will be used by the server side of your app. You can generate them in the Firebase console under Project settings > Service accounts > Firebase Admin SDK > Generate new private key.
+
+### Setting up credentials
+
+To add your Firebase credentials to Astro, create an `.env` file in the root of your project with the following variables:
+
+```ini title=".env"
+FIREBASE_PRIVATE_KEY_ID=YOUR_SPACE_ID
+FIREBASE_PRIVATE_KEY=YOUR_DELIVERY_TOKEN
+FIREBASE_PROJECT_ID=YOUR_PREVIEW_TOKEN
+```
+
+Now, you should be able to use these environment variables in your project. 
+
+If you would like to have IntelliSense for your Firebase environment variables, create an `env.d.ts` file in the `src/` directory and configure your types like this:
+
+```ts title="src/env.d.ts"
+interface ImportMetaEnv {
+  readonly FIREBASE_PRIVATE_KEY_ID: string;
+  readonly FIREBASE_PRIVATE_KEY: string;
+  readonly FIREBASE_PROJECT_ID: string;
+}
+
+interface ImportMeta {
+  readonly env: ImportMetaEnv;
+}
+```
+
+:::tip
+Read more about [environment variables](/en/guides/environment-variables/) and `.env` files in Astro.
+:::
+
+Your root directory should now include these new files:
+
+<FileTree title="Project Structure">
+- src/
+  - **env.d.ts**
+- **.env**
+- astro.config.mjs
+- package.json
+</FileTree>
+
+
+### Installing dependencies
+
+To connect Astro with Firebase, install the following packages using the single command below for your preferred package manager:
+
+- `firebase` - the Firebase SDK for the client side
+- `firebase-admin` - the Firebase Admin SDK for the server side
+
+<PackageManagerTabs>
+  <Fragment slot="npm">
+  ```shell
+  npm install firebase firebase-admin
+  ```
+  </Fragment>
+  <Fragment slot="pnpm">
+  ```shell
+  pnpm install firebase firebase-admin
+  ```
+  </Fragment>
+  <Fragment slot="yarn">
+  ```shell
+  yarn install firebase firebase-admin
+  ```
+  </Fragment>
+</PackageManagerTabs>
+
+Next, create a folder named `firebase` in the `src/` directory. 
+Inside the firebase folder create two files, `client.ts` and `server.ts`.
+
+The `client.ts` file will contain the code to initialize Firebase in the client using the web app credentials.
+
+```ts title="src/firebase/client.ts"
+import { initializeApp } from "firebase/app";
+
+const firebaseConfig = {
+  apiKey: "my-public-api-key",
+  authDomain: "astro-firebase.firebaseapp.com",
+  projectId: "astro-firebase",
+  storageBucket: "astro-firebase.appspot.com",
+  messagingSenderId: "469295531959",
+  appId: "1:469295531959:web:20b9159723f8113f2744f2",
+};
```

Comment:
Doesn't it make sense to also refer to these with environment variables. If not, why?